### PR TITLE
AttachmentProperty features

### PR DESCRIPTION
**Pull request by wilcoschoneveld**
Right now the attachment property has the following functions:

```
property.value = matplotlib.figure  # upload figure (plot) to KE-chain
property.upload('test.csv')   # upload a local file called test.csv to KE-chain
property.save_as('something.txt')  # download the file from KE-chain to a local file 
```

There is not yet clear functionality to handle json formats, although a lot is possible out of the box with requests.

@jberends what features are required and do you want to say something related to unifying the attachment property API?

### CODE REVIEW

**Comment by jberends** [comment]
I would say

- (chunked) upload any attachment file (bound to accepted mime types
- (chunked) download any attachment file (using the /download api detail route) such that it serves as the correct mime

File: pykechain/models/attachment.py [hunk 1]
```diff
@@ -26,6 +26,15 @@ def value(self, value):
         except ImportError:
             pass
 
+    def upload(self, filename):
+        """Upload a file to the attachment property.
+
+        :param filename: File path
+        :raises: APIError
+        """
+        with open(filename, 'rb') as f:
```
Comment (jberends) [inline pykechain/models/attachment.py]: add assert / os.path.exist checks
add FileNotFoundError raising

Comment (wilcoschoneveld) [inline pykechain/models/attachment.py]: `open` already facilitates this :stuck_out_tongue_winking_eye:

**Comment by wilcoschoneveld** [comment]
#22 take over

**Pull request closed without merging**